### ISSUE TITLE
BlueSky crash on start - resolved

### ISSUE THREAD

**Issue by tinpusher** [TUDelft-CNS-ATM/bluesky#507]
Had an issue when starting BlueSky, would not boot through.
error log ... 'np.string_' was removed in the NumPy 2.0 release. Use 'np_bytes_' instead.

I removed Numpy2.0, and used :  pip install numpy==1.26.4 , dated 06/02/2024.
This seems to have fixed the booting issue.

**Comment by thomasrdean** [comment]
I tried that, but I get 
Bluesky needs numpy._core.numeric
And 
pip install numpy._core.numeric
says it could not find a version that satisfies the requirements numpy._core.numeric

**Comment by jooste** [comment]
Hi @thomasrdean and @tinpusher! This seems to be an issue introduced with the new major release of NumPy. We'll have a look at what needs to be changed in BlueSky to be compatible with these changes.

**Comment by tinpusher** [comment]
It seems that Qt gui threw a fatal error.
radarwidget.py line 138,  in initializeGL super().initializeGL
glhelpers.py line 752,  in initializeGL obj.creater()
navdata.py,line 123, in create npwids=np.array(wptids, dtype=np.string_.
Seems that to support both 1.x and 2.x versions of NumPy,  modules will need to be compiled with numPy 2.0.
....
I took the easy way out.... downgraded to the last 1.x NumPy version I found ...1.26.4

**Comment by thomasrdean** [comment]
When I downgraded, I got the error message about core.numeric.

**Comment by VYalla** [comment]
Has anyone found a temporary working solution? Encountering the same issue, and seeing the issue has been titled as resolved.

I tried downgrading numpy to 1.26.4, however getting same error message about core.numeric as @thomasrdean

**Comment by tinpusher** [comment]
Maybe check if numpy2.0, or some remaining artifacts are not in the path, uninstall then reinstall the previous version, ensuring it is in the path. I used pip to remove and reinstall the previous version. This was all that I needed to do. When pip reinstalls the previous version, it automatically uninstalls the newer numpy version.

**Comment by VYalla** [comment]
Yes, I tried that again (uninstalling numpy 2.0 via _pip uninstall numpy -y_, and clearing from the path, and reinstalling with 1.26.4 via _pip install numpy==1.26.4_), however still running into the same issue when running _python3 BlueSky.py_:

```
C:\Users\StuCheckout\Documents\Vishal\bluesky>python3 BlueSky.py
Using Python-based geo functions
Reading config from C:\Users\StuCheckout\Documents\Vishal\bluesky\settings.cfg
Reading magnetic variation data
Loading global navigation database...
Reading cache: C:\Users\StuCheckout\Documents\Vishal\bluesky\cache\navdata.p
Bluesky needs numpy._core.numeric
Run setup-python.bat (Windows) or check requirements.txt (other systems)
Install using e.g. pip install numpy._core.numeric
BlueSky normal end.
```

Current numpy version: 
```
>python -c "import numpy; print(numpy.__version__)"
1.26.4
```

**Comment by VYalla** [comment]
The solution that worked for me was simply changing the following lines in `\bluesky\bluesky\ui\qtgl\glnavdata.py` to the following (simply changing the dtype from np.string_ to np.bytes_ respectively):

Line 123: 
```
npwpids = np.array(wptids, dtype=np.bytes_)
```
Line 158:
```
self.aptlabels.create(np.array(aptids, dtype=np.bytes_),
                              self.airports.lat, self.airports.lon,
                              palette.aptlabel, (apt_size, 0.5 * apt_size),
                              instanced=True)

```

This worked for me with NumPy 2.0.

**Comment by Kriso23** [comment]
> The solution that worked for me was simply changing the following lines in `\bluesky\bluesky\ui\qtgl\glnavdata.py` to the following (changed the dtype from np.string_ to np.bytes_ respectively):
> 
> Line 123:
> 
> ```
> npwpids = np.array(wptids, dtype=np.bytes_)
> ```
> 
> Line 158:
> 
> ```
> self.aptlabels.create(np.array(aptids, dtype=np.bytes_),
>                               self.airports.lat, self.airports.lon,
>                               palette.aptlabel, (apt_size, 0.5 * apt_size),
>                               instanced=True)
> ```
> 
> This worked for me with NumPy 2.0.

This is enough to start the program however I needed to change 2 more np.string_ to np.bytes_ (basically all np.string_) and then I was able to create aircraft. Hopefully I won't encounter any other issues.

**Comment by guessh0o** [comment]
> > The solution that worked for me was simply changing the following lines in `\bluesky\bluesky\ui\qtgl\glnavdata.py` to the following (changed the dtype from np.string_ to np.bytes_ respectively):
> > Line 123:
> > ```
> > npwpids = np.array(wptids, dtype=np.bytes_)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Line 158:
> > ```
> > self.aptlabels.create(np.array(aptids, dtype=np.bytes_),
> >                               self.airports.lat, self.airports.lon,
> >                               palette.aptlabel, (apt_size, 0.5 * apt_size),
> >                               instanced=True)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > This worked for me with NumPy 2.0.
> 
> This is enough to start the program however I needed to change 2 more np.string_ to np.bytes_ (basically all np.string_) and then I was able to create aircraft. Hopefully I won't encounter any other issues.

Hi, may I know which other 2 np.string_? manage to find 1 more only

I am having this issue on my macOS and it was fine 2 weeks ago, I am getting "Bluesky needs numpy._core.numeric". Please share if you manage to solve it, thanks!

**Comment by Kriso23** [comment]
> > > The solution that worked for me was simply changing the following lines in `\bluesky\bluesky\ui\qtgl\glnavdata.py` to the following (changed the dtype from np.string_ to np.bytes_ respectively):
> > > Line 123:
> > > ```
> > > npwpids = np.array(wptids, dtype=np.bytes_)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Line 158:
> > > ```
> > > self.aptlabels.create(np.array(aptids, dtype=np.bytes_),
> > >                               self.airports.lat, self.airports.lon,
> > >                               palette.aptlabel, (apt_size, 0.5 * apt_size),
> > >                               instanced=True)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > This worked for me with NumPy 2.0.
> > 
> > 
> > This is enough to start the program however I needed to change 2 more np.string_ to np.bytes_ (basically all np.string_) and then I was able to create aircraft. Hopefully I won't encounter any other issues.
> 
> Hi, may I know which other 2 np.string_? manage to find 1 more only
> 
> I am having this issue on my macOS and it was fine 2 weeks ago, I am getting "Bluesky needs numpy._core.numeric". Please share if you manage to solve it, thanks!

If you change the following np.string_ to np.bytes_ it should work (at least for me so far i have no problems)
![image](https://github.com/user-attachments/assets/a8530cf8-a318-4a80-a02d-277d681643a6)

**Comment by jooste** [comment]
Hi all,

Thanks for the suggestions. This has been fixed in 56ae5de.
